core/src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala

```diff
@@ -152,6 +153,89 @@ Expected DataFrame Row Count: '$expectedCount'
     }
   }
 
+  /**
+   * Raises an error unless `actualDS` and `expectedDS` are equal. It is recommended to provide `primaryKeys` to ensure accurate and efficient
+   * comparison of rows. If primary key is not provided, will try to compare the rows based on their row number. This requires both datasets to be
+   * partitioned in the same way and become unreliable when shuffling is involved.
+   * @param primaryKeys
+   *   unique identifier for each row to ensure accurate comparison of rows
+   * @param checkKeyUniqueness
+   *   if true, will check if the primary key is actually unique
+   */
+  def assertLargeDatasetEqualityV2[T: ClassTag](
```

Comment:
How should we approach this? Should I add more version that is closer to what we had before? or maybe just use this replace older version with this

Reply:
We can mark an old one as deprecated.